### PR TITLE
 Allowed to return null for query_builder

### PR DESCRIPTION
**Pull request by JonEastman**
| Q | A |
| --- | --- |
| Doc fix? | yes |
| New docs? | no |
| Applies to | 2.8 |
| Fixed tickets | #5934 |

### CODE REVIEW

**Comment by HeahDude** [comment]
Thanks, can you please update [this line too](https://github.com/symfony/symfony-docs/pull/6594/files#diff-2a47ce41986258d5e75671c0c20be152R200)?

File: reference/forms/types/entity.rst [hunk 1]
```diff
@@ -202,9 +202,9 @@ query_builder
 Allows you to create a custom query for your choices. See
 :ref:`ref-form-entity-query-builder` for an example.
 
-The value of this option can either be a ``QueryBuilder`` object or a Closure.
+The value of this option can either be a ``QueryBuilder`` object, a Closure or null.
```
Comment (WouterJ) [inline reference/forms/types/entity.rst]: unless I'm missing something, it's not the option value that can be `null`. Instead, the closure can return null.

File: reference/forms/types/entity.rst [hunk 2]
```diff
@@ -202,9 +202,9 @@ query_builder
...
 When using a Closure, you will be passed the ``EntityRepository`` of the entity
-as the only argument and should return a ``QueryBuilder``.
+as the only argument and should return a ``QueryBuilder``.  If you'd like to display a list of empty entries, you can return null in the query_builder closure.
```
Comment (WouterJ) [inline reference/forms/types/entity.rst]: as we're in the `query_builder` documentation here, the word `query_builder` can be removed from "the query_builder closure".

Can you please also put `null` in double backticks (so it's displayed as an inline code block) and add a line break after the first word that crosses the 72th character?

**Comment by WouterJ** [comment]
Hi @JonEastman! Thanks for your contribution, I've left 2 minor comments.

File: reference/forms/types/entity.rst (hunk 1 above)
Comment (HeahDude) [inline reference/forms/types/entity.rst]: @WouterJ see https://github.com/symfony/symfony/blob/2.3/src/Symfony/Bridge/Doctrine/Form/Type/DoctrineType.php#L167 and https://github.com/symfony/symfony/blob/2.7/src/Symfony/Bridge/Doctrine/Form/Type/EntityType.php#L43 but this change should be done in 2.3 though

**Comment by JonEastman** [comment]
@WouterJ I made the suggested changes, let me know what you think

**Comment by WouterJ** [comment]
:+1: Looks good!

**Comment by xabbuh** [comment]
:+1:

File: reference/forms/types/entity.rst [hunk 3]
```diff
@@ -202,9 +202,10 @@ query_builder
 Allows you to create a custom query for your choices. See
 :ref:`ref-form-entity-query-builder` for an example.
 
-The value of this option can either be a ``QueryBuilder`` object, a Closure or null.
+The value of this option can either be a ``QueryBuilder`` object, a Closure or ``null``.
 When using a Closure, you will be passed the ``EntityRepository`` of the entity
-as the only argument and should return a ``QueryBuilder``.  If you'd like to display a list of empty entries, you can return null in the query_builder closure.
+as the only argument and should return a ``QueryBuilder``.  
+If you'd like to display a list of empty entries, you can return ``null`` in the closure.
```
Comment (HeahDude) [inline reference/forms/types/entity.rst]: Unless I'm missing something when `null` is returned it loads all entities.

File: reference/forms/types/entity.rst [hunk 4]
```diff
@@ -202,9 +202,10 @@ query_builder
 Allows you to create a custom query for your choices. See
 :ref:`ref-form-entity-query-builder` for an example.
 
-The value of this option can either be a ``QueryBuilder`` object, a Closure or null.
+The value of this option can either be a ``QueryBuilder`` object, a Closure or ``null``.
```
Comment (HeahDude) [inline reference/forms/types/entity.rst]: Needs a rebase here when #6596 is merged in 2.8

Comment (WouterJ) [inline reference/forms/types/entity.rst]: I'll take care of that while merging.

**Comment by WouterJ** [comment]
Thanks @JonEastman! I've merged your PR in the documentation and added a little versionadded box indicating that this is a new feature in f64fe19 (fixed the version in a later commit).

File: reference/forms/types/entity.rst (hunk 3 above)
Comment (xabbuh) [inline reference/forms/types/entity.rst]: When you set the option to `null`, but not when return `null` in the closure, isn't it?

Comment (HeahDude) [inline reference/forms/types/entity.rst]: Why that? The only difference is that it's conditional. ?

Comment (xabbuh) [inline reference/forms/types/entity.rst]: I don't think so (see https://github.com/symfony/symfony/pull/13990/files#diff-888009597299c21d535422fd4368bed3R72).

Comment (HeahDude) [inline reference/forms/types/entity.rst]: Ok so I've missed the point of that feature, thanks for the enlightening!

Comment (HeahDude) [inline reference/forms/types/entity.rst]: @xabbuh Looks like https://github.com/symfony/symfony/commit/317d30b6efd4e906e9c84fbe558934448379285c has not made it to the core !? (neither [master](https://github.com/symfony/symfony/blob/master/src/Symfony/Bridge/Doctrine/Form/ChoiceList/ORMQueryBuilderLoader.php#L50) nor [2.8](https://github.com/symfony/symfony/blob/2.8/src/Symfony/Bridge/Doctrine/Form/ChoiceList/ORMQueryBuilderLoader.php#L84))

Comment (WouterJ) [inline reference/forms/types/entity.rst]: Looks like there was something wrong with the commit and it was fixed in https://github.com/symfony/symfony/commit/eb08baa9fcb6db3863c795ea54e324ced68d56ea

The tests exists and passes: https://github.com/symfony/symfony/blob/master/src/Symfony/Bridge/Doctrine/Tests/Form/Type/EntityTypeTest.php#L212-L223

Comment (HeahDude) [inline reference/forms/types/entity.rst]: see https://github.com/symfony/symfony/pull/13990#issuecomment-220788142

**Pull request closed without merging**